### PR TITLE
[resource_container_cluster] - Add self-link property

### PR DESCRIPTION
**Pull request by ScottSuarez**
Add self link property to resource_container_cluster. 

Resolves [7120](https://github.com/hashicorp/terraform-provider-google/issues/7120)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Added self_link to google_container_cluster
```

### CODE REVIEW

File: third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -817,6 +817,12 @@ func resourceContainerCluster() *schema.Resource {
 				Description:      `The name or self_link of the Google Compute Engine subnetwork in which the cluster's instances are launched.`,
 			},
 
+			"self_link": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Server-defined URL for the resource.`,
+			}
```
Comment (megan07) [inline third_party/terraform/resources/resource_container_cluster.go.erb:824]: I think we want a comma after this here.

**Review by megan07 — CHANGES_REQUESTED** [review]
Hi @ScottSuarez! Were you able to build this locally? If you need any help with that, feel free to reach out!

**Review by megan07 — APPROVED** [review]
Looks good! Thanks for cleaning up the whitespace. Do you think we should add a test in for this?

**Comment by ScottSuarez** [comment]
> Looks good! Thanks for cleaning up the whitespace. Do you think we should add a test in for this?

It certainly couldn't hurt! I added the property validation to an existing basic test and ran it locally.

**Pull request merged**